task/prefetch-dependencies/0.1/prefetch-dependencies.yaml

```diff
@@ -125,6 +137,10 @@ spec:
       Workspace containing a .netrc file. Cachi2 will use the credentials in this file when
       performing http(s) requests.
     optional: true
+  - name: prefetch-config
+    description: |
+      Workspace containing a cachi2 YAML config file to further tweak its behaviour.
```

Comment:
Passing the config as a workspace may not be ideal. The user would essentially have to create a ConfigMap or a Secret in their namespace and pass it to the pipeline as a workspace. At which point the behavior of the build is not configurable in the repo, it depends on an external Kubernetes object.

IMO it would be preferable to do one of
* pass the content of the config file as a param
* pass the path to the config file as a param, the user commits it into their repo (although at that point it would make more sense if the user could also incorporate (an equivalent of) the JSON input for fetch-deps into that file)
* pass the config as an array param of `NAME=value` strings, the task converts them to environment variables (after https://github.com/containerbuildsystem/cachi2/issues/573)

Reply:
> pass the config as an array param of NAME=value strings, the task converts them to environment variables (after https://github.com/containerbuildsystem/cachi2/issues/573)

^This was my intention all along, this PR is supposed to be just a workaround (or a band aid if you will) to unblock existing consumers, so I'm wondering if for the time being (i.e. before https://github.com/containerbuildsystem/cachi2/issues/573 lands) the consumers could live with a temporary suboptimal user experience, or whether we do want to spend more time with this, especially since with my nonexistent knowledge of Konflux I can't objectively estimate how much work would the following pose:

> pass the content of the config file as a param

Reply:
Easier than the workspace solution, conceptually something like this

```diff
diff --git a/task/prefetch-dependencies/0.1/prefetch-dependencies.yaml b/task/prefetch-dependencies/0.1/prefetch-dependencies.yaml
index b993d3e..91970f5 100644
--- a/task/prefetch-dependencies/0.1/prefetch-dependencies.yaml
+++ b/task/prefetch-dependencies/0.1/prefetch-dependencies.yaml
@@ -12,6 +12,8 @@ spec:
     Task that uses Cachi2 to prefetch build dependencies.
     See docs at https://github.com/containerbuildsystem/cachi2#basic-usage.
   params:
+  - name: config-file-content
+    description: cachi2 config file content
   - description: Configures project packages that will have their dependencies prefetched.
     name: input
   - description: >
@@ -50,6 +52,8 @@ spec:
       value: $(workspaces.netrc.bound)
     - name: WORKSPACE_NETRC_PATH
       value: $(workspaces.netrc.path)
+    - name: CONFIG_FILE_CONTENT
+      value: $(params.config-file-content)
     volumeMounts:
       - name: trusted-ca
         mountPath: /mnt/trusted-ca
@@ -68,6 +72,13 @@ spec:
         dev_pacman_flag=""
       fi
 
+      if [ -n "$CONFIG_FILE_CONTENT" ]; then
+        printf "%s" "$CONFIG_FILE_CONTENT" > /tmp/cachi2.yaml
+        configfile_flag=--config-file=/tmp/cachi2.yaml
+      else
+        configfile_flag=''
+      fi
+
       # Copied from https://github.com/konflux-ci/build-definitions/blob/main/task/git-clone/0.1/git-clone.yaml
       if [ "${WORKSPACE_GIT_AUTH_BOUND}" = "true" ] ; then
         if [ -f "${WORKSPACE_GIT_AUTH_PATH}/.git-credentials" ] && [ -f "${WORKSPACE_GIT_AUTH_PATH}/.gitconfig" ]; then

```

Reply:
lovely, thank you @acmiel let me steal and apply that!